datamade/la-metro-councilmatic
# Upload analytics report to Google Drive

## Overview

Generates an analytics CSV file and uses a Google Service Account to write it to a pre-defined folder in Google Drive. I held off on sharing the folder with LA Metro colleagues until this has been tested.

### Checklist



## Testing Instructions

 * Run `blackbox_postdeploy` to decrypt the config files.
 * Get the ID of a test Google Drive folder you'd like to use. See note below for instructions.
 * Add `REMOTE_ANALYTICS_FOLDER = 'id'` (where id is the ID of the folder) to the end of `councilmatic/settings_deployment.py`.
 * Run `docker-compose run --rm app python manage.py generate_tag_analytics`
 * Verify that there is a new file in your test folder.
 * Run `blackbox_shred_all_files` to remove decrypted files from your local repo.


### Notes

In a browser, you can find the ID of a Google Drive folder by right-clicking it and selecting 'Get link':

![id](https://user-images.githubusercontent.com/36973363/160923641-5f86b0c5-91e3-4fb9-85ce-d867bb5ff468.jpg)

The ID is the highlighted portion of the URL in the picture above.

Handles #814
